lib/elixir/src/elixir_interpolation.erl

```diff
@@ -78,9 +78,24 @@ unescape_tokens(Tokens) ->
   unescape_tokens(Tokens, fun unescape_map/1).
 
 unescape_tokens(Tokens, Map) ->
-  [unescape_token(Token, Map) || Token <- Tokens].
```

Comment:
This is one of the few cases where I think a `throw` would be clearer. So something like this:

```erlang
unescape_tokens(Tokens, Map) ->
  try [unescape_token(Token, Map) || Token <- Tokens] of
    Unescaped -> {ok, Unescaped}
  catch
    {error, _Msg} = Error -> Error
  end.
```

This way you can keep the implementation the same, all you need to do is to replace the ArgumentError by a throw.